lib/licensed/shell.rb

```diff
@@ -3,16 +3,14 @@
 
 module Licensed
   module Shell
-    # Executes a command, returning its standard output on success. On failure,
-    # it raises an exception that contains the error output, unless
-    # `allow_failure` is true, in which case it returns an empty string.
+    # Executes a command, returning its standard output on success.
+    # On failure it raises an exception that contains the error output, unless
+    # `allow_failure` is true.
     def self.execute(cmd, *args, allow_failure: false, env: {})
       stdout, stderr, status = Open3.capture3(env, cmd, *args)
 
-      if status.success?
+      if status.success? || allow_failure
```

Comment:
allowing a failure will now return the command output for a failed command.  this was necessary, for example, for the NPM source to find dependencies even if some couldn't be found and the NPM CLI exited non-zero.